src/main/java/com/ibm/cloudant/kafka/connect/CloudantSourceTask.java

```diff
@@ -42,210 +40,218 @@
+            password = config.getPassword(InterfaceConst.PASSWORD).value();
+            topics = config.getList(InterfaceConst.TOPIC);
+            taskNumber = config.getInt(InterfaceConst.TASK_NUMBER);
+            generateStructSchema = config.getBoolean(InterfaceConst.USE_VALUE_SCHEMA_STRUCT);
+
+            latestSequenceNumber = config.getString(InterfaceConst.LAST_CHANGE_SEQ);
+            batch_size = config.getInt(InterfaceConst.BATCH_SIZE) == null ? InterfaceConst
+                    .DEFAULT_BATCH_SIZE : config.getInt(InterfaceConst.BATCH_SIZE);
 				
 			/*if (tasks_max > 1) {
-				throw new ConfigException("CouchDB _changes API only supports 1 thread. Configure tasks.max=1");
+				throw new ConfigException("CouchDB _changes API only supports 1 thread. Configure
+				tasks.max=1");
 			}*/
-												
-			_running = new AtomicBoolean(false);
-			stop = new AtomicBoolean(false);
-			
-			if (latestSequenceNumber == null) {
-				latestSequenceNumber = new String(DEFAULT_CLOUDANT_LAST_SEQ);
-				
-				OffsetStorageReader offsetReader = context.offsetStorageReader();
-
-				if (offsetReader != null) {
-					Map<String, Object> offset = offsetReader.offset(Collections.singletonMap(InterfaceConst.URL, url));
-					if (offset != null) {
-						latestSequenceNumber = (String) offset.get(InterfaceConst.LAST_CHANGE_SEQ);
-						LOG.info("Start with current offset (last sequence): " + latestSequenceNumber);
-					}
-				}
-			}
-
-			// Create a new CloudantClient instance for account endpoint account.cloudant.com
-			String urlWithoutProtocal = url.substring(url.indexOf("://") +3);
-			String account = urlWithoutProtocal.substring(0,urlWithoutProtocal.indexOf("."));
-			
-			cantClient = ClientBuilder.account(account)
-			                          .username(userName)
-			                          .password(password)
-			                          .build();
-			
-			// Create a database instance
-			cantDBName = url.substring(url.lastIndexOf("/")+1);
-			
-			// Create a database instance
-			cantDB = cantClient.database(cantDBName, false);
-			
-		} catch (ConfigException e) {
-			throw new ConnectException(ResourceBundleUtil.get(MessageKey.CONFIGURATION_EXCEPTION), e);
-		}
-
-	}
-
-	@Override
-	public void stop() {
-		if (stop != null) {
-			stop.set(true);
-		}
-		
-		// terminate the changes feed
-		// Careful: this is an asynchronous call
-		if (cantDB.changes() != null) {		
-			cantDB.changes().stop();
-			
-			// graceful shutdown
-			// allow the poll() method to flush records first
-			if (_running == null) return;
-			
-			while (_running.get() == true) {
-				try {
-					Thread.sleep(SHUTDOWN_DELAY_MILLISEC);
-				} catch (InterruptedException e) {
-					LOG.error(e);
-				}
-			}
-		}
-	}
-	
+
+            _running = new AtomicBoolean(false);
+            stop = new AtomicBoolean(false);
+
+            if (latestSequenceNumber == null) {
+                latestSequenceNumber = new String(DEFAULT_CLOUDANT_LAST_SEQ);
```

Comment:
I believe the new String(...) is redundant and we can just do `latestSequenceNumber = DEFAULT_CLOUDANT_LAST_SEQ`.

Reply:
Fixed in 7c205d5719d12ff03091d5149405206e2ae498d5